http-parser/src/http_parser.mli

```diff
@@ -0,0 +1,24 @@
+(** Attempts to parse a buffer into a HTTP request. If successful, it returns
+    the parsed request and an offset value that indicates the starting point of
+    unconsumed content left in the buffer. *)
+
+type error = Partial | Msg of string
+type chunk_kind = Start_chunk | Continue_chunk of int
+
+type chunk_parser_result =
+  | Chunk_complete of string
+  | Done
+  | Partial_chunk of string * int
+
+val parse_request :
+  ?pos:int -> ?len:int -> string -> (Http.Request.t * int, error) result
+
+val parse_chunk_length :
+  ?pos:int -> ?len:int -> string -> (int * int, error) result
```

Comment:
I have used  `int` for Chunk lengths as the initial work that led to this parser made an assumption that we'll always run on 64bit systems. With that assumption the `int` would be enough to represent the largest chunk size that's allowed by the parser.